File: src/main/java/com/jacuum/engine/IterationEvent.java
package com.jacuum.engine;

import com.jacuum.algo.Direction;

public record IterationEvent(
    String sessionId,
    int iteration,
    Direction direction,
    int robotX,
    int robotY,
    int score,
    int totalCleaned,
    int totalFloor,
    boolean finished,
    FinishReason finishReason
) {}

File: src/main/kotlin/com/jacuum/engine/IterationEvent.kt
package com.jacuum.engine

import com.jacuum.algo.Direction

data class IterationEvent(
    val sessionId: String,
    val iteration: Int,
    val direction: Direction?,
    val robotX: Int,
    val robotY: Int,
    val score: Int,
    val totalCleaned: Int,
    val totalFloor: Int,
    val finished: Boolean,
    val finishReason: FinishReason?
)
